src/clients/SpokePoolClient.ts

```diff
@@ -656,6 +656,21 @@ export class SpokePoolClient extends BaseAbstractClient {
         }
 
         if (this.depositHashes[getRelayEventKey(deposit)] !== undefined) {
+          // Sanity check that this event is not a duplicate, even though the relay data hash is a duplicate.
+          const allDeposits = this._getDuplicateDeposits(deposit).concat(this.depositHashes[getRelayEventKey(deposit)]);
+          if (
+            allDeposits.some((e) => {
+              return e.transactionHash === deposit.transactionHash && e.logIndex === deposit.logIndex;
+            })
+          ) {
+            this.logger.warn({
```

Comment:
If this is ever logged, this basically means that there is a bug somewhere. I think this should be an error.

Reply:
this might take down the fast relayer though, no?

Reply:
I think the relayers are the only bots which do not stop on errors, since they error a lot when txns revert in simulation. @pxrl could probably confirm that I'm not mistaken, though.

Reply:
I'm not sure we should crash here. I'd rather just ignore the event and try our best to  keep the events stored in memory correct by filtering out duplicates

Reply:
Ideally, this should never trigger, though. I'd argue that if this is logged, then we introduced a bug somewhere and should look into recent commits. Right now, we know that this would occasionally log in the relayer, but if this happened somewhere else in the future, we'd probably want to look into it.

Reply:
hm so if we crashed here, this would bring down the fast relayer which seems to handlese duplicate events ok today. Are we ok with that? yes it'd help us debug, but also be disprutive to production service. WDYT?

Reply:
If the fast relayer crashes on errors, then I agree, we should not log an error and go with what you currently have. I don't _think_ the fast relayer crashes on logged errors, though. Am I wrong?

Reply:
ce0be1dae58f9d634803ea60161943ea236682ea